sync/sync.go

```diff
@@ -265,10 +265,18 @@ func (s *Synchronizer) syncBlocks(syncCtx context.Context) {
 	}
 }
 
+func maxWorkers() int {
+	m, mProcs := 16, runtime.GOMAXPROCS(0)
+	if mProcs > m {
+		return m
+	}
+	return mProcs
+}
+
 func (s *Synchronizer) setupWorkers() (*stream.Stream, *stream.Stream) {
 	numWorkers := 1
 	if s.catchUpMode {
-		numWorkers = runtime.GOMAXPROCS(0)
+		numWorkers = maxWorkers()
 	}
 	return stream.New().WithMaxGoroutines(numWorkers), stream.New().WithMaxGoroutines(numWorkers)
```

Comment:
Hmm, contrary to what I have done before. I don't think we need to limit `verifierTask` limit.
```suggestion
	return stream.New().WithMaxGoroutines(numWorkers), stream.New().WithMaxGoroutines(runtime.GOMAXPROCS(0))
```

If fetcherTask (even with 1 worker) manages to fetch multiple blocks faster than we should be able to verify them in parallel.

Reply:
Done